src/d/actor/d_a_kita.cpp

```diff
@@ -4,52 +4,448 @@
+            
+            float fVar4;
+            short local29A;
+            if(i_this->field_360 == 2){
+                fVar4 = 1.0;
+                local29A = 800;    
+            }
+            else {
+                fVar4 = 0.0;
+                local29A = 2500;
+            }
+            i_this->field_2C8.x = i_this->field_2BC.x * JMASSin(i_this->mMoveCounter * local29A);
+            i_this->field_2C8.z = i_this->field_2BC.z * JMASSin(i_this->mMoveCounter * (local29A - 200));
+            cLib_addCalc2(&i_this->field_2BC.x, (REG0_F(9) + 400.0f) * fVar4, 1.0, REG0_F(3) + 20.0f);
+            cLib_addCalc2(&i_this->field_2BC.z, (REG0_F(9) + 400.0f) * fVar4, 1.0, REG0_F(3) + 20.0f);
+            
+            i_this->shape_angle = i_this->current.angle + i_this->field_2C8;
+            i_this->current.pos.y += i_this->speed.y;
+            i_this->speed.y -= 5.0f;
+            if(i_this->speed.y < -150.0f) i_this->speed.y = -150.0f;
+            
+            if(i_this->current.pos.y <= i_this->field_35C){
+                i_this->current.pos.y = i_this->field_35C;
+                if(i_this->speed.y < -50.0f){
+                    i_this->field_2BC.z= 2000.0f;
+                    i_this->field_2BC.x = 2000.0f;
+                    if(i_this->field_360 != 0){
+                        i_this->field_360 = 2;
+                        fopAcM_seStart(i_this, 0x6953, 0);
+                        cXyz particle_scale(3.0, 3.0, 3.0);
+                        dComIfGp_particle_set(0x828C, &i_this->current.pos);
+                        dComIfGp_particle_set(0x3F, &i_this->current.pos, 0, &particle_scale);
+
+                        dComIfGp_getVibration().StartShock(REG0_S(2) + 4, -0x21, cXyz(0.0, 1.0, 0.0));
+                    }
+                    else{
+                        fopAcM_seStart(i_this, 0x6951, 0);
+                    }
+                }
+                i_this->speed.y = 0;
+                cLib_addCalcAngleS2(&i_this->mRotX, 0, 2, 0x2000);
+            }
+            if(i_this->field_360 != 0){
+                i_this->mAcch.CrrPos(*dComIfG_Bgsp());
+            }
+            break;
+    }
 }
 
 /* 00001894-000019F8       .text himo_create__FP10kita_class */
-cPhs_State himo_create(kita_class*) {
-    /* Nonmatching */
+cPhs_State himo_create(kita_class* i_this) {
+    /* Nonmatching - yad does not have the right address in .data section */
+    static short yad[4] = {0x2000, 0xE000, 0x6000, 0xA000};
+    int shand_count = 0;
+    fopAcM_prm_class *param;
+    fopAc_ac_c* shand_i;
+
+    for(int i = 0; i < 4; i++){
+        switch(i_this->field_2E4[i]){
+            case 0:
+                param = fopAcM_CreateAppend();
+                param->base.position = i_this->current.pos;
+                param->base.angle.y = yad[i];
+                param->base.parameters = 0xFFFFFF35;
+                param->room_no = i_this->current.roomNo;
+                i_this->field_2D4[i] = fpcM_Create(PROC_SHAND, NULL, param);
```

Comment:
```suggestion
                i_this->field_2D4[i] = fopAcM_Create(PROC_SHAND, NULL, param);
```
They usually prefer fopAcM inlines over fpcM inlines in actor code for semantic reasons (no practical difference afaik)